src/titiler/mosaic/titiler/mosaic/factory.py

```diff
@@ -659,6 +666,17 @@ def wmts(
                         </TileMatrix>"""
                 tileMatrix.append(tm)
 
+            if use_epsg:
+                supported_crs = f"EPSG:{tms.crs.to_epsg()}"
+            else:
+                supported_crs = tms.crs.srs
+
+            bbox_crs_type = "WGS84BoundingBox"
+            bbox_crs_uri = "urn:ogc:def:crs:OGC:2:84"
+            if tms.rasterio_geographic_crs != WGS84_CRS:
+                bbox_crs_type = "BoundingBox"
+                bbox_crs_uri = CRS_to_uri(tms.rasterio_geographic_crs)
```

Comment:
here an above, I think that uri's are not the same as urn's (functionally very similar) although I forget if qgis/gdal was fine with either representation. I also made this code addition as a feature branch a while ago to morecantile that could be used but don't want to block this pr https://github.com/developmentseed/morecantile/pull/163/files

Reply:
to be honest I've read the WMTS spec and it's not clear what the `URI` should be, I feel it would be fine as long that pyproj/rasterio can resolve `CRS.from_user_input(uri)`

Reply:
okay great https://github.com/cogeotiff/rio-tiler/pull/752 is up, will need to look again at https://github.com/rasterio/rasterio/pull/3206 soon though to get my datasets supported though (hopefully would be a patch level fix...)